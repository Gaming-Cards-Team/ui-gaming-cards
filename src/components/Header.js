import React from "react";
import logo from "../assets/images/logo.png";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";
import { LOGO_ALTERNATIVE_TEXT } from "../util/constant";

const Header = () => {
  return (
    <Navbar bg="light" variant="light">
      <Navbar.Brand href="#home">
        <img
          data-testid="logo"
          height="48"
          src={logo}
          alt={LOGO_ALTERNATIVE_TEXT}
        />
      </Navbar.Brand>
    </Navbar>
  );
};

export default Header;
