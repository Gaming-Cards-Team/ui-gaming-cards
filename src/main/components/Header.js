import AppBar from "@material-ui/core/AppBar";
import { makeStyles } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import React from "react";
import logo from "../resources/logo.png"

const useStyles = makeStyles({
  root: {
    flexGrow: 1
  },
  logo: {
    width: 135,
    height: 43.54
  }
});

const Header = () => {
  const classes = useStyles();
  return (
    <nav className={classes.root} id='header'>
      <AppBar position="static" color="default">
        <Toolbar>
          <img
            data-testid='logo'
            className={classes.logo}
            src={logo}
            alt="Level Up Logo"
          />
        </Toolbar>
      </AppBar>
    </nav>
  );
};

export default Header;
