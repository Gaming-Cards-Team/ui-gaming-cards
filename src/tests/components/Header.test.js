import React from 'react';
import Header from '../../main/components/Header';
import { shallow } from 'enzyme';

describe('Test Header component', () => {

    it('should have a header div', () => {
        
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });

});