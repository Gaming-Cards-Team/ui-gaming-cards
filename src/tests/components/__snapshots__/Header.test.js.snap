// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component should validate the Header component 1`] = `
<nav
  className="navbar navbar-expand navbar-light bg-light fixed-top"
>
  <a
    className="navbar-brand"
    href="#home"
  >
    <img
      alt="Level Up Logo"
      className=""
      data-testid="logo"
      height="48"
      src="logo.png"
    />
  </a>
</nav>
`;
